CASMINST-7035: Improve "SAT and IUF" section

haasken-hpe opened the pull request:
# Description
Improve the "SAT and IUF" section of the documentation that describes the relationship between SAT and IUF, specifically how `sat bootprep` is executed in IUF and how variable substitution works in that context.

Include a procedure to obtain the `session_vars.yaml` file and bootprep input files from the IUF activity and use them to create CFS configurations, images, and session templates manually. This can be a useful debugging procedure.

Include another procedure that describes how to assign the CFS configurations and IMS images to management NCNs as a non-standard debugging procedure. This procedure used to be used when performing an upgrade of only CSM on a system that also contains other products that were previously installed by IUF.

Move a section about limiting `sat bootprep run` into separate stages into the `SAT_Bootprep.md` file because it's really more appropriate with the general `sat bootprep` usage information.

# Checklist

- [x] If I added any command snippets, the steps they belong to follow the prompt conventions (see [example][1]).
- [x] If I added a new directory, I also updated `.github/CODEOWNERS` with the corresponding team in [Cray-HPE][2].
- [x] My commits or Pull-Request Title contain my JIRA information, or I do not have a JIRA.

<!--- These are Markdown Reference Style URLs, they do not show in the PR --> 
[1]: https://github.com/Cray-HPE/docs-csm/blob/main/introduction/documentation_conventions.md#using-prompts
[2]: https://github.com/Cray-HPE/teams

leliasen-hpe commented on operations/system_admin_toolkit/usage/SAT_and_IUF.md:
````diff
@@ -168,57 +141,46 @@ example showing how to find the IUF activity.
+   
    ```bash
-   yq w -i "${BOOTPREP_DIR}/session_vars.yaml" 'csm.working_branch' "integration-${CSM_RELEASE}"
-   yq r "${BOOTPREP_DIR}/session_vars.yaml" 'csm.working_branch'
+   cd "${BOOTPREP_DIR}"
+   sat bootprep run --vars-file session_vars.yaml management-bootprep.yaml
    ```
````
Should this piece be put in the next section? It seem more applicable to building images and cfs configs than it does to "Obtaining IUF session variables and bootprep files".

haasken-hpe replied:
Yeah, I think that's fair. I just wanted to make sure it was clear what you would do with these files in this section to actually use them in bootprep.

I will move it perhaps with a comment about generally how these files would be used.

haasken-hpe commented:
Thanks for your review and comment, @leliasen-hpe. I added some separate commits addressing some issues I noticed while reviewing and addressing your comment. I also had to address some spelling and style issues. Please take another look, and if it looks good, please let me squash my fixup commits into the original commit and then I'll merge into your branch if that's all right.

leliasen-hpe commented on operations/system_admin_toolkit/usage/SAT_Bootprep.md:
```diff
@@ -448,40 +448,69 @@ session_templates:
+
+This section describes how to view the variables defined by a given installed
+version of the HPC CSM Software Recipe. As described in [Variable Substitutions](#variable-substitutions),
+these variables must be rendered by the IUF before they can be used by `sat
```
```suggestion
these variables must be rendered by IUF before they can be used by `sat
```

haasken-hpe replied:
Do you think it should be "IUF" instead of "the IUF" everywhere?

haasken-hpe replied:
I made the change from "the IUF" to "IUF" everywhere IUF is used as a noun by itself.

leliasen-hpe commented on operations/system_admin_toolkit/usage/SAT_and_IUF.md:
```diff
@@ -7,106 +7,80 @@ to use `sat bootprep`.
+
+As described in [Variable substitutions](SAT_Bootprep.md#variable-substitutions),
+the `sat bootprep` command supports variable substitutions in the `bootprep` input
+files. The IUF merges variables from multiple sources and then passes the resulting
```
```suggestion
files. IUF merges variables from multiple sources and then passes the resulting
```

leliasen-hpe commented on operations/system_admin_toolkit/usage/SAT_and_IUF.md:
```diff
@@ -7,106 +7,80 @@ to use `sat bootprep`.
+the `sat bootprep` command supports variable substitutions in the `bootprep` input
+files. The IUF merges variables from multiple sources and then passes the resulting
+merged variables to `sat bootprep` using the `--vars-file` option. The sources
+used by the IUF, in order of lowest to highest precedence are as follows:
```
This is a nit and it is fine to be left as is. In the other document you did highest to lowest precedence and here you do the opposite. I had to think a tiny bit about it but really it works as it.

haasken-hpe replied:
Darn, you noticed! 😆 Which order do you think is more natural? High precedence to low or low to high?

haasken-hpe replied:
I went high to low here to match the other location.

leliasen-hpe commented on operations/system_admin_toolkit/usage/SAT_and_IUF.md:
```diff
@@ -7,106 +7,80 @@ to use `sat bootprep`.
+
+### Obtaining IUF session variables and bootprep files
+
+When the IUF constructs the variables as described in [IUF Variable
```
```suggestion
When IUF constructs the variables as described in [IUF Variable
```
